tests/test_super.rs

```diff
@@ -0,0 +1,52 @@
+#![cfg(feature = "macros")]
+
+use pyo3::prelude::*;
+
+#[pyclass(subclass)]
+struct BaseClass {
+    val1: usize,
+}
+
+#[pymethods]
+impl BaseClass {
+    #[new]
+    fn new() -> Self {
+        BaseClass { val1: 10 }
+    }
+
+    pub fn method1(&self) -> usize {
+        self.val1
+    }
+}
+
+#[pyclass(extends=BaseClass)]
+struct SubClass {}
+
+#[pymethods]
+impl SubClass {
+    #[new]
+    fn new() -> (Self, BaseClass) {
+        (SubClass {}, BaseClass::new())
+    }
+
+    fn method2<'a>(self_: PyRef<'_, Self>, py: Python<'a>) -> PyResult<&'a PyAny> {
+        let any: Py<PyAny> = self_.into_py(py);
+        let super_ = any.into_ref(py).py_super()?;
```

Comment:
Think this can just be written as
```suggestion
        let super_ = self_.py_super()?;
```

Reply:
@davidhewitt I correctly understood that I need to add new method to `PyRef`?

Reply:
Ah right, yes `PyRef` doesn't deref to `PyAny`. `PyCell<T>` does, so I think let's instead go with:

```rust
    fn method<'py>(self_: &'py PyCell<Self>) -> PyResult<&'py PyAny> {
         let super_ = self_.py_super()?;
```

Reply:
Done!